### PR TITLE
[ENH] pivot_wider names_glue

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- names_glue is now a string, and allows for easier interpolation, similar to tidyr's pivot_wider names_glue
- levels_order is redundant and excised
- edited docs
- more heavy name modification can be handled by the user via sort_index/swaplevel/reorder_levels/collapse_levels

**This PR improves pivot_wider.**

# PR Checklist

Please ensure that you have done the following:

1. [ ] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [ ] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [ ] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1052--pyjanitor.netlify.app

**Review by ericmjl — APPROVED** [review]

File: tests/functions/test_conditional_join.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def test_single_condition_less_than_ints_extension_array(df, right):
     left_on, right_on = ["A", "Integers"]
     expected = (
         df.merge(right, how="cross")
-        .query(f"{left_on} < {right_on}")
+        .query(f"{left_on} < {right_on}", engine="python")
```
Comment (ericmjl) [inline tests/functions/test_conditional_join.py]: Curious, @samukweku, according to the `pandas` [docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html):

> You can change the semantics of the expression by passing the keyword argument parser='python'. This enforces the same semantics as evaluation in Python space. Likewise, you can pass engine='python' to evaluate an expression using Python itself as a backend. This is not recommended as it is inefficient compared to using numexpr as the engine.

Would it be more advantageous if we left the `engine` kwarg to be the default `numexpr`?

**Review by ericmjl — APPROVED** [review]
Just one question, @samukweku. Pre-approving, though!

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,8 +20,9 @@
 -   [BUG] Fix SettingWithCopyWarning and other minor bugs when using `truncate_datetime_dataframe`, along with further performance improvements (PR #1040). @thatlittleboy
 -   [ENH] Performance improvement for `conditional_join`. @samukweku
 -   [ENH] Multiple `.value` is now supported in `pivot_longer`. #1034 @samukweku
--   [ENH] Add `xlsx_cell` for reading a spreadsheet as a table of individual cells. #929 @samukweku.
--   [EHN] Let `filter_string` suit parameters of `Series.str.contains` Issue #1003 and #1047. @Zeroto521
+-   [ENH] Add `xlsx_cells` for reading a spreadsheet as a table of individual cells. #929 @samukweku.
+-   [ENH] Let `filter_string` suit parameters of `Series.str.contains` Issue #1003 and #1047. @Zeroto521
+-   [ENH] `names_glue` now takes a string form, using str.format_map under the hood. `levels_order` is also deprecated. @samukweku
```
Comment (thatlittleboy) [inline CHANGELOG.md]: Worth mentioning this is for the `pivot_wider` function here?

**Review by thatlittleboy — APPROVED** [review]
Same question as Eric, and 1 small change requested for Changelog.
Otherwise LGTM!

File: tests/functions/test_conditional_join.py (hunk 1 above)
Comment (samukweku) [reply tests/functions/test_conditional_join.py]: numexpr for some reason was failing, hence the `engine=python` fix. Made the changes, using pandas' loc instead. I'll make the changes for the others in another PR.

**Pull request merged**